app/main.py

```diff
@@ -1,2 +1,100 @@
+from typing import Any
+from typing import Iterator
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.initial_capacity = initial_capacity
+        self.buckets = [{} for _ in range(initial_capacity)]
+
+    @property
+    def _load_factor(self) -> float:
+        return len(self) / self.initial_capacity
+
+    def _resize(self) -> None:
+        if self._load_factor > 2 / 3:
+            new_size = self.initial_capacity * 2
+            new_buckets = [{} for _ in range(new_size)]
+
+            for bucket in self.buckets:
+
+                for key, value in bucket.items():
+                    new_hash_key = hash(key) % new_size
+                    new_buckets[new_hash_key][key] = value
+            self.initial_capacity = new_size
+            self.buckets = new_buckets
+
+    def __setitem__(self, key: int, value: str) -> None:
+        self._resize()
+        hash_key = self._hash_function(key)
+        bucket = self.buckets[hash_key]
+        if key in bucket:
+            bucket[key] = value
+        else:
+            bucket[key] = value
+
+    def __getitem__(self, key: Any) -> Any:
+        hash_key = self._hash_function(key)
+        if key in self.buckets[hash_key]:
+            return self.buckets[hash_key][key]
+        else:
+            raise KeyError(key)
+
+    def __len__(self) -> int:
+        return sum(len(bucket) for bucket in self.buckets)
+
+    def __list__(self) -> Any:
+        data_list = []
+        for bucket in self.buckets:
+            data_list.extend(bucket)
+        return data_list
+
+    def clear(self) -> None:
+        for bucket in self.buckets:
+            bucket.clear()
+
+    def pop(self, object_: dict, default_value: Any = None) -> Any:
+        bucket_index = self._get_bucket_index(object_)
+        bucket = self.buckets[bucket_index]
+
+        for i, (stored_key, value) in enumerate(bucket):
+            if stored_key == object_:
+                del bucket[i]
+                return value
+
+        if default_value is not None:
+            return default_value
```

Comment:
But FYI:
d = {}
d.pop(1) # Raise KeyError
d.pop(1, None) # Not Raise KeyError